### ISSUE TITLE
Underscore disappear

### ISSUE THREAD

**Issue by hdsuperman** [ajaxorg/ace#4000]
## Underscore disappear in ace editor (at right)

chrome: (left: in kitchen-sink , right: in real environment)
![Screenshot from 2019-07-01 19-51-11](https://user-images.githubusercontent.com/50064165/60434555-f50f4b80-9c39-11e9-8eeb-f4c897772d73.png)

firefox is ok:
![Screenshot from 2019-07-01 19-58-06](https://user-images.githubusercontent.com/50064165/60434854-b29a3e80-9c3a-11e9-9340-528d653f2b85.png)

opera has error too:
![Screenshot from 2019-07-01 19-58-24](https://user-images.githubusercontent.com/50064165/60434947-eaa18180-9c3a-11e9-88c7-eb3e17e1bc49.png)

code :
```
<template>
  <div></div>
</template>

<script lang="ts">

  import {Vue, Component, Prop} from "vue-property-decorator";
  import ace from 'ace-builds';
  import 'ace-builds/webpack-resolver';
  import 'ace-builds/src-noconflict/theme-github';
  import 'ace-builds/src-noconflict/mode-markdown';

  @Component
  export default class SuCodeEditor extends Vue {
    @Prop({type: Number, default: 10}) minLines!: number;

    @Prop({type: Number, default: 20}) maxLines!: number;

    @Prop({type: Number, default: 14}) fontSize!: number;

    @Prop({type: Boolean, default: true}) showGutter!: boolean;

    aceEditor: any = null;
    themePath: string = 'ace/theme/github';
    modePath: string  = 'ace/mode/markdown';

    mounted() {
      this.aceEditor = ace.edit(this.$el, {
        behavioursEnabled: true,
        highlightActiveLine: false,
        showGutter: this.showGutter,


        maxLines: this.maxLines, // 最大行数，超过会自动出现滚动条
        minLines: this.minLines, // 最小行数，还未到最大行数时，编辑器会自动伸缩大小
        fontSize: this.fontSize,            // 编辑器内字体大小
        theme: this.themePath,   // 默认设置的主题
        mode: this.modePath,     // 默认设置的语言模式
        tabSize: 4               // 制表符设置为 4 个空格大小

      });
    }
  }
</script>

<style scoped>

</style>

```

**Comment by nightwing** [comment]
Which font are you using?

does it help if you add one of the following css snippets?
```css
.ace_editor {
    line-height: 2;
}
```
or 
```css
.ace_editor * {
    contain: none!important; 
}
```

**Comment by albfan** [comment]
I think this is related https://github.com/node-red/node-red/issues/2104#issuecomment-559188837

As soon as you zoom-in underscore is seen:
- 100%: ![100% ace editor](https://user-images.githubusercontent.com/220968/69745235-1bcaa300-1142-11ea-864e-a8c9c0a3b1d5.png)
- 110%: ![110%](https://user-images.githubusercontent.com/220968/69745261-27b66500-1142-11ea-8e37-352810f60ad9.png)

font-family for ACE editor is:
```
font-family: Menlo,Consolas,"DejaVu Sans Mono",Courier,monospace !important;
```
If you do not have that fonts installed it is not seen. As soon as you remove that property underscore is seen.

With that property it renders to Arial:

![arial rendering](https://user-images.githubusercontent.com/220968/69746936-b37dc080-1145-11ea-8d88-df0336a9a58d.png)

Removing the property it renders to Ubuntu mono here:
![Captura de pantalla de 2019-11-27 18-40-18](https://user-images.githubusercontent.com/220968/69746994-ce503500-1145-11ea-9a37-dc2125069c8c.png)

So basically, it is using a no monospace font, that renders behind the bottom line.

I found a similar report here:
https://github.com/rust-lang/mdBook/issues/1061

**Comment by olliewd** [comment]
Hi, I had this issue too - at least when using the react-ace wrapper. I was able to get round it by setting the lineHeight sufficiently high. For the font I was using (Droid Sans Mono), any line height below 18px wouldn't render the underscores at default zoom. However, with line heights above 18px, the underscores would always show, no matter how zoomed you were. These values are likely to vary from font to font. I have no clue why this is the case, but it appears to work.

**Comment by olliewd** [comment]
d

**Comment by albfan** [comment]
Yep, see this fix on node-red

https://github.com/node-red/node-red/pull/2579/files

**Comment by andrewnester** [comment]
Is this still reproducible on the latest Ace version?
